rewrite-java-test/src/test/java/org/openrewrite/java/ReplaceStringLiteralWithConstantTest.java

```diff
@@ -19,103 +19,253 @@
 import org.openrewrite.test.RecipeSpec;
 import org.openrewrite.test.RewriteTest;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.openrewrite.java.Assertions.java;
 
 class ReplaceStringLiteralWithConstantTest implements RewriteTest {
     @Override
     public void defaults(RecipeSpec spec) {
         spec
           .parser(JavaParser.fromJavaVersion()
-            .logCompilationWarningsAndErrors(true)
-            .classpath("guava"));
+            .logCompilationWarningsAndErrors(true));
+    }
+
+    @Test
+    void shouldReadValuesInOrderProperties() {
+        ReplaceStringLiteralWithConstant recipe = (ReplaceStringLiteralWithConstant) RecipeSpec.defaults().recipeFromYaml("""
+            type: specs.openrewrite.org/v1beta/recipe
+            name: org.openrewrite.ReplaceStringLiteralWithConstantList
+            recipeList:
+                - org.openrewrite.java.ReplaceStringLiteralWithConstant:
+                    literalValue: %s
+                    fullyQualifiedConstantName: %s
+            """.formatted("default", EXAMPLE_STRING_FQN),
+          "org.openrewrite.ReplaceStringLiteralWithConstantList"
+        ).getRecipe().getRecipeList().get(0).getRecipeList().get(0);
+        assertThat(recipe.getFullyQualifiedConstantName()).isEqualTo(EXAMPLE_STRING_FQN);
+        assertThat(recipe.getLiteralValue()).isEqualTo("default");
```

Comment:
I understand your concerns about the test readability here; I don't know of any other constructs though and appreciate that you've taken the time to verify that the values match what's (not) configured here.

Reply:
The only different approach I could think of, was converting this test case into a rewriteRun and just assert code changes. Then, we would implicitly assert the default handling. The current code shows more obviously what the intention behind the test case but the alternative would be more blackbox-ish and thus more robust. What do you think about this?

Reply:
Converting them into tests that actually run could work as well; then we cover the most likely way to use this recipe. Optional to have that replace some of the explicit constructor calls if you feel we're covering things twice or more.

Reply:
I decided to keep constructor calls as main strategy to create the recipes, because that is how it is done in other test classes. Then, I refactored the test cases around yaml to also execute rewrite runs and asserted the expected changes.